src/components/CookieConsent/CookieConsent.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+import {useMobile} from '@gravity-ui/uikit';
+
+import {block} from '../utils/cn';
+
+import {ConsentMode, Consents} from './ConsentManager';
+import {ConsentNotification} from './components/ConsentNotification/ConsentNotification';
+import {ConsentPopup} from './components/ConsentPopup/ConsentPopup';
+import {SimpleConsent} from './components/SimpleConsent/SimpleConsent';
+import {CookieConsentProps} from './types';
+
+const b = block('analytics');
+
+export const CookieConsent = ({
+    consentManager,
+    onConsentPopupClose,
+    disableInitialOpen,
+    forceOpen,
+    ...popupProps
+}: CookieConsentProps) => {
+    const [isOpened, setIsOpened] = React.useState(false);
+    const [mobile] = useMobile();
+
+    React.useEffect(() => {
+        // Show banner after some timeout so that the user has time to see the service content
+        const timeoutId = setTimeout(() => {
+            if (!disableInitialOpen && consentManager.isConsentNotDefined()) {
+                setIsOpened(true);
+            }
+        }, 1000);
+
+        return () => clearTimeout(timeoutId);
+    }, [disableInitialOpen, consentManager]);
+
+    const onConsentPopupAction = (consents: Consents) => {
+        consentManager.setConsents(consents);
+        setIsOpened(false);
```

Comment:
`onAction` is always called with `onClose`

Reply:
add onConsentPopupClose in onConsentPopupAction